pkg/apis/sources/v1/sinkbinding_lifecycle.go

```diff
@@ -102,6 +102,7 @@ func (sb *SinkBinding) Do(ctx context.Context, ps *duckv1.WithPod) {
 	resolver := GetURIResolver(ctx)
 	if resolver == nil {
 		logging.FromContext(ctx).Errorf("No Resolver associated with context for sink: %+v", sb)
+		return
```

Comment:
I think it's better than a panic. Though now it's a silent failure. I am curious if in both of these resolve failures (nil and uri resolution error) we should be setting the status to failed with the reason. Right now these are not surfaced except in the logs, seems better to sb.Status.MarkSinkFailed(...). Thoughts?

Reply:
completely agreed, didn't even think of that in my initial approach, will start drafting that up now!